src/lib/cached.ts

```diff
@@ -188,7 +188,7 @@ export const cached = {
                 });
         },
     },
-};
+} as const;
```

Comment:
Since this was declared const, does this `as const` do some magic I don't know about or is it redundant?

Reply:
Yep a bit of magic :)

The key difference with the declarations is that the type is not widened when using `as const`[^1]:

```
const obj = { foo: "foo"; bar: 28 } // type is { foo: string; bar: number }
const obj_as_const = { foo: "foo"; bar: 28 } // type is { foo: "foo"; bar: 28 }
```

This is important because data now only accepts keys that have been listed in the schema, so lines like this would fail if `cached.challenge_list` were an arbitrary `string`:

https://github.com/online-go/online-go.com/blob/48efeca3e5120f5fb63a8a9af2e901d1fe6bdf37/src/lib/cached.ts#L72

[^1]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions

Reply:
BTW we *could* allow data to accept arbitrary strings, but I'm not sure that's desirable.  I'm thinking about potential mistakes like:

```
data.get("confgi.user");
              ^^
             typo
```

strict key checking could be quite useful!

Reply:
Fancy, yeah strict sounds good :)